tests/core/Observer.spec.ts

```diff
@@ -329,23 +408,43 @@ describe('Observer', async () => {
           'transactionNumber': 2,
           'transactionTime': 2001,
           'transactionTimeHash': '2001',
-          'anchorString': '2ndTransactionNew',
+          'anchorString': AnchoredDataSerializer.serialize({
+            anchorFileHash: 'hash2New',
+            numberOfOperations: 1
+          }),
           'transactionFeePaid': 1,
           'normalizedTransactionFee': 1
         },
         {
           'transactionNumber': 3,
           'transactionTime': 3001,
           'transactionTimeHash': '3000',
-          'anchorString': '3rdTransactionNew',
+          'anchorString': AnchoredDataSerializer.serialize({
+            anchorFileHash: 'hash3New',
+            numberOfOperations: 1
+          }),
           'transactionFeePaid': 1,
           'normalizedTransactionFee': 1
         },
         {
           'transactionNumber': 4,
           'transactionTime': 4000,
           'transactionTimeHash': '4000',
-          'anchorString': '4thTransaction',
+          'anchorString': AnchoredDataSerializer.serialize({
```

Comment:
These changes are not really needed anymore right? I mean the observer tests now only need to spy on the througputLimiter call it seems.

Reply:
Do you not think it's useful to test end to end?

Reply:
E2E tests are definitely useful but we are not doing end 2 end in these tests are we?

My concern is that if we change some logic in the throughputLimiter, we would have to come back and change these tests. IMO, moving that logic out of that class (or even into a new function) has the additional advantage of making our testing easier. A new test can be added which tests whether the througputLimiter is being called or not and only the txns returned by that function are being processed.

Given above, if you still want to keep these here then I'm fine as you have already done the hard work and written the tests.